IPython/extensions/autoreload.py

```diff
@@ -552,21 +605,45 @@ def autoreload(self, parameter_s=""):
           autoreloaded.
 
         """
-        if parameter_s == "":
+        args = magic_arguments.parse_argstring(self.autoreload, line)
+        mode = args.mode.lower()
+
+        def p(msg):
+            print(msg)
+
+        def l(msg):
+            logging.getLogger("autoreload").info(msg)
+
+        def pl(msg):
+            p(msg)
+            l(msg)
+
+        if args.print is False and args.log is False:
+            self._reloader._report = lambda msg: None
+        elif args.print is True:
+            if args.log is True:
+                self._reloader._report = lambda msg: pl(msg)
```

Comment:
Unless there is a reason for wrapping in lambdas I would suggest to just assign the function directly.

```suggestion
                self._reloader._report = pl
```